frontend/NextJs/style-guide.md

```diff
@@ -0,0 +1,219 @@
+  as request parameters.<br /><br />
+
+  ```js
+  function MyComponent(props) {
+    return (
+      // Use the retrieved data as component props
+      <div>{props.data}</div>
+    )
+  }
+
+  export async function getServerSideProps(context) {
+    // Retrieve data from theexternal API
+    const res = await fetch.get('https://mi-api.com/datos', {
+      params: context.query,
+    })
+    const data = res.data
+
+    // Pass the data as props to the component
+    return {props: {data}}
+  }
+  ```
+
+In addition to **getStaticProps** and **getServerSideProps**, in Next.js you can
+also use other external tools and libraries to fetch data, such as react-query
+or the Fetch API. You can also use the React useEffect function to fetch data
+when a component is mounted, and then store the data in the component's state so
+that it can be used in your user interface.
+
+## API Routes
+
+In Next.js API routes are a way to build server-side logic for handling
+asynchronous requests in a Next.js application. They allow you to create
+endpoints that can be used to retrieve or update data, perform tasks, or
+communicate with external APIs. These routes are typically used to handle
+server-side logic and are not intended to be used for rendering traditional web
+pages. API routes in Next.js are implemented as special files in the /pages/api
+directory of your application. Each file in this directory represents a separate
+API endpoint and can export a default function that will be invoked when a
+request is made to that endpoint. For example, you might create an /api/users
+endpoint to handle requests related to retrieving, updating, and deleting user
+data. This endpoint could then be used to perform database queries or call
+external APIs to retrieve the necessary data and return it to the client in a
+JSON. Here is an example of an API route in Next.js that returns a list of users
+in JSON format: // pages/api/users.js
+
+```js
+import users from './data/users.json'
+
+export default (req, res) => {
+  res.status(200).json(users)
+}
+```
+
+This API route exports a default function that is invoked when a request is made
+to the /api/users endpoint. The function uses the res object to send a JSON
+response to the client with a list of users. You can then make a request to this
+endpoint from your front-end code using a tool like fetch. For example:
+
+```js
+fetch('/api/users')
+  .then(res => res.json())
+  .then(data => console.log(data))
+```
+
+This will send a GET request to the /api/users endpoint and log the response
+data to the console.
+
+You can read more examples
+[here](https://github.com/vercel/next.js/tree/canary/examples/api-routes-rest)
+You can read the docs [here](https://nextjs.org/docs/api-routes/introduction)
+
+## SEO
+
+In Next.js you can improve the SEO of the websites using some
+[pre-rendering modes](#fetching-data) this modes will we could to get a document
```

Comment:
I think that the redaction here is a little bit messy, don't know if I'm getting the right idea, how about?
```suggestion
[pre-rendering modes](#fetching-data) these modes will generate the application
```

Reply:
Hi @MMarussi! what about `these modes will let you get an optimized HTML document`?

Reply:
Perfect!